search-app/src/cmr/search/api/concepts_search.clj

```diff
@@ -5,17 +5,27 @@
    [cmr.common-app.api.routes :as common-routes]
    [cmr.common-app.services.search :as search]
    [cmr.common.cache :as cache]
+   [cmr.common.config :refer [defconfig]]
    [cmr.common.log :refer (debug info warn error)]
    [cmr.common.mime-types :as mt]
    [cmr.common.services.errors :as svc-errors]
+   [cmr.common.util :as util]
    [cmr.search.api.core :as core-api]
    [cmr.search.services.parameters.legacy-parameters :as lp]
    [cmr.search.services.query-service :as query-svc]
    [cmr.search.services.result-format-helper :as rfh]
+   [cmr.search.validators.all-granule-validation :as all-gran-validation]
    [compojure.core :refer :all]))
 
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;;; Constants and Utility Functions
+(defconfig allow-all-granule-params-flag
+  "Flag that indicates if we allow all granule queries."
+  {:default true :type Boolean}) 
```

Comment:
This should default to `false`.

Reply:
I mentioned it in the scrum this morning that defaulting it to false caused too many errors - over a thousand because we have too many existing all granule searches...   I've just added some tests to test the case when it's set to false.

Reply:
Now that I fixed a couple of bugs, there should be a little less, but still a lot. Also we will have to re-design the tests, for example, currently one test throws an error saying the concept-id is not valid, if I change the default to false, the failure is at the all granule case is not allowed. Where are we going to put concept-id test?  We could work around it by setting the allow-all-granule-params-flag to true just for those cases, but there are too many of them.

Reply:
I think you can put `cmr-allow-all-granule-params-flag true` in profiles.clj to override the default config in `concept-service` in local dev. In CI build, set an env variable. We use this approach to set things like db password right now.